fanduelOdds.php

```diff
@@ -35,13 +94,218 @@ private function getLiveMarketIds() {
          *
          * @return array Array of live market ids
          */
-    }
 
-    private function getMarketPrices() {
+        // setup request to ListMarketCatalogue API
+        $liveMarketsRequestPayload = json_encode([
+            "listMarketCatalogueRequestParams" => [
+                "marketFilter" => [
+                    "eventTypeIds" => [self::BASKETBALL_ID],
+                    "competitionIds" => [self::NBA_ID],
+                    "marketTypes" => self::MARKET_TYPES
+                ],
+                "maxResults" => 1000
+            ]
+        ]);
+        
+        $requestHeaders = [
+            "Content-type: application/json",
+            "X-Application: " . $this->secret_key
+        ];
+
+        $liveMarketsCh = curl_init();
+
+        curl_setopt_array($liveMarketsCh, [
+            CURLOPT_URL => $this->liveMarketsEndpoint,
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_CUSTOMREQUEST => "POST",
+            CURLOPT_POSTFIELDS => $liveMarketsRequestPayload,
+            CURLOPT_HTTPHEADER => $requestHeaders
+        ]);
+
+        $liveMarketsResponse = curl_exec($liveMarketsCh);
+
+        // process the response
+        if (curl_errno($liveMarketsCh)) {
+            echo 'Error (getLiveMarketIds): ' . curl_error($liveMarketsCh);
+            curl_close($liveMarketsCh);
+            return [];
+        } 
+        else {
+            $data = json_decode($liveMarketsResponse);
+            curl_close($liveMarketsCh);
+
+            $liveMarketIds = [];
+            // grab all liveMarketIds
+            foreach ($data as $item) {
+                array_push($liveMarketIds, $item->marketId);
+            }
+            return $liveMarketIds;
+        }
+    }
+    private function getMarketPrices($liveMarketIds) {
         /**
          * Helper function that returns market prices for Spread Betting, Money Line and Total Points for next Raptors game
          *
          * @return array Array of market prices for next Raptors game
          */
+        
+        // setup request to ListMarketPrices API
+        $marketPricesRequestPayload = json_encode([
+            "listMarketPricesRequestParams" => [
+                "marketIds" => $liveMarketIds,
+            ]
+        ]);
+
+        $requestHeaders = [
+            "Content-type: application/json",
+            "X-Application: " . $this->secret_key
+        ];
+    
+        $marketPricesCh = curl_init();
```

Comment:
Same point as before, if you able to extract the curl calling into its own method, you can re-use it here instead of duplicating.